File: android/app/src/main/java/com/kufarcode/app/domain/model/City.kt
package com.kufarcode.app.domain.model

data class City(
    val code: String,
    val name: String,
    val theme: CityTheme,
    val features: CityFeatureFlags
)

data class CityTheme(
    val primaryColor: String,
    val secondaryColor: String? = null,
    val logoUrl: String? = null
)

data class CityFeatureFlags(
    val enableMap: Boolean = true,
    val enableFavorites: Boolean = true,
    val experimentalLayout: Boolean = false
)

File: mobile/src/types/city.ts
export interface CityTheme {
  primaryColor?: string;
  logoUrl?: string;
}

export interface CityFeatures {
  enableFarmers?: boolean;
  enableCrafts?: boolean;
  enableTaxi?: boolean;
  [key: string]: boolean | undefined;
}

export interface CityConfig {
  code: string;
  name: string;
  theme?: CityTheme;
  features?: CityFeatures;
}
